File: Baito/Baito/src/main/java/Baito/order/service/OrderService.java
package Baito.order.service;

import Baito.order.dto.CreateOrderRequest;
import Baito.order.model.Order;
import Baito.product.product.Product;

import java.util.List;

public interface OrderService {

    
    List<Order> getUserOrders(Long userId);

    List<Order> getAdminOrders(String filter);

    Order updateStatus(Long orderId, String status);

    Order createOrder(CreateOrderRequest request, Long userId);
    // Add product to existing order
    // Increase product quantity in order
    Order increaseQuantity(Long orderId, Long productId, Integer quantityToAdd);

    // Decrease product quantity in order
    Order decreaseQuantity(Long orderId, Long productId, Integer quantityToReduce);

}

File: frontend/src/services/orderService.js
import { fetchAPI } from './api';

export const orderService = {
  // Create order with items
  createOrder: async (username, phone, address, items) => {
    return fetchAPI('/orders/create', {
      method: 'POST',
      body: JSON.stringify({
        username,
        phone,
        address,
        items: items.map(item => ({
          productId: item.productId,
          quantity: item.quantity,
        })),
      }),
    });
  },

  // Get user's orders
  getMyOrders: async () => {
    return fetchAPI('/orders/my', { method: 'GET' });
  },

  // Get all orders (admin)
  getAllOrders: async (filter = null) => {
    const url = filter ? `/orders/all?filter=${filter}` : '/orders/all';
    return fetchAPI(url, { method: 'GET' });
  },

  // Update order status (admin)
  updateOrderStatus: async (orderId, status) => {
    return fetchAPI(`/orders/${orderId}/status?status=${status}`, {
      method: 'PUT',
    });
  },

  // Increase product quantity
  increaseQuantity: async (orderId, productId, quantity) => {
    return fetchAPI(
      `/orders/${orderId}/product/${productId}/increase?quantity=${quantity}`,
      { method: 'PUT' }
    );
  },

  // Decrease product quantity
  decreaseQuantity: async (orderId, productId, quantity) => {
    return fetchAPI(
      `/orders/${orderId}/product/${productId}/decrease?quantity=${quantity}`,
      { method: 'PUT' }
    );
  },
};

export default orderService;